pkg/client/database/timescale/timescale.go

```diff
@@ -97,3 +113,39 @@ func (e Executor) Finalize(ctx context.Context, retErr error) error {
 
 	return retErr // keep migrate.ErrNoChange
 }
+
+func runningTimescaleVersion(ctx context.Context, q queryRower) (*version.Version, error) {
+	row := q.QueryRow(ctx, timescaleVersionQuery)
+	var result string
+	if err := row.Scan(&result); err != nil {
+		return nil, err
+	}
+	fields := strings.Fields(result)
+	if len(fields) != 1 {
+		return nil, fmt.Errorf("failed to parse version: %s, %q", result, fields)
+	}
+	return version.NewVersion(fields[0])
+}
+
+// ValidateTimescaleVersion checks that Timescale plugin version available through pool is not lower than wanted version.
+// In this case it returns nil. Otherwise returns error describing current and desired version or any other error encountered
+// during the check.
+func ValidateTimescaleVersion(ctx context.Context, pool *pgxpool.Pool, want *version.Version) error {
+	conn, err := pool.Acquire(ctx)
+	if err != nil {
+		return err
+	}
+	defer conn.Release()
+	return doValidateTimescaleVersion(ctx, conn, want)
+}
+
+func doValidateTimescaleVersion(ctx context.Context, q queryRower, want *version.Version) error {
+	got, err := runningTimescaleVersion(ctx, q)
+	if err != nil {
+		return fmt.Errorf("error getting Timescale version: %w", err)
+	}
+	if got.LessThan(want) {
+		return fmt.Errorf("unsupported Timescale version: %v. (should be >= %v)", got, want)
```

Comment:
not sure if needed, but:

```suggestion
		return fmt.Errorf("unsupported Timescale version: %s. (should be >= %s)", got.String(), want.String())
```

Reply:
Fixed in both TS and PG